store/src/java-test/com/zimbra/cs/service/mail/ModifyFilterRulesTest.java

```
@@ -660,7 +660,53 @@ public void testFilterVariablesWithNoValue() {
+            Element rule = rules.addElement(MailConstants.E_FILTER_RULE);
+            rule.addAttribute(MailConstants.A_ACTIVE, true);
+            Element filteraction = rule.addElement(MailConstants.E_FILTER_ACTIONS);
+            Element actionInto = filteraction.addElement(MailConstants.E_ACTION_TAG);
+            actionInto.addAttribute(MailConstants.A_TAG_NAME, "123-456");
+            filteraction.addElement(MailConstants.E_ACTION_STOP);
+
+            Element filterTests = rule.addElement(MailConstants.E_FILTER_TESTS);
+            // Not set the "condition=" parameter.
+
+            Element headerTest1 = filterTests.addElement(MailConstants.E_HEADER_TEST);
+            headerTest1.addAttribute(MailConstants.A_HEADER, "subject");
+            headerTest1.addAttribute(MailConstants.A_STRING_COMPARISON, "contains");
+            headerTest1.addAttribute(MailConstants.A_VALUE, "123");
+
+            Element headerTest2 = filterTests.addElement(MailConstants.E_HEADER_TEST);
+            headerTest2.addAttribute(MailConstants.A_HEADER, "X-Header");
+            headerTest2.addAttribute(MailConstants.A_STRING_COMPARISON, "contains");
+            headerTest2.addAttribute(MailConstants.A_VALUE, "456");
+
+            new ModifyFilterRules().handle(request, ServiceTestUtil.getRequestContext(acct));
+
+            String expectedScript = "require [\"fileinto\", \"reject\", \"tag\", \"flag\", \"variables\", \"log\", \"enotify\"];\n\n";
+            expectedScript += "# null\n";
+            expectedScript += "if allof (header :contains [\"subject\"] \"123\",\n";
+            expectedScript += "  header :contains [\"X-Header\"] \"456\") {\n";
+            expectedScript += "    tag \"123-456\";\n";
+            expectedScript += "    stop;\n";
+            expectedScript += "}\n";
+
+            assertEquals(expectedScript, acct.getMailSieveScript());
+
+        } catch (ServiceException e) {
```

Putting all of this in a try/catch block will hide the original exception stack from the unittest report.

Better to just mark the test as `throws Exception`